File: 360-video-player-for-android-master/app/src/main/java/com/oculus/sample/ContactsAdapter.java
package com.oculus.sample;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;


public class ContactsAdapter extends ArrayAdapter {

    List list = new ArrayList();

    public ContactsAdapter(Context context, int resource) {
        super(context, resource);
    }


    public void add(Contacts object) {
        super.add(object);
        list.add(object);
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Nullable
    @Override
    public Object getItem(int position) {
        return list.get(position);
    }



    @NonNull
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
            View row;
            row = convertView;
        ContactHolder contactHolder;
        if(row==null){
            LayoutInflater layoutInflater=(LayoutInflater)this.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
                row = layoutInflater.inflate(R.layout.row_layout,parent,false);
                        contactHolder = new ContactHolder();
            contactHolder.Item=(TextView)row.findViewById(R.id.item);
                row.setTag(contactHolder);
        }else{

           contactHolder = (ContactHolder)row.getTag();
        }

        Contacts contacts =(Contacts)this.getItem(position);
        contactHolder.Item.setText(contacts.getItem());
        return row;
    }



    static class ContactHolder
    {
        TextView Item;

    }
}

File: 360-video-player-for-android-master/app/src/main/java/com/oculus/sample/ContactsAdapter.kt
package com.oculus.sample

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class ContactsAdapter(context: Context, resource: Int) : ArrayAdapter<Any?>(context, resource) {

    private val list: MutableList<Any?> = ArrayList()

    fun add(obj: Contacts) {
        super.add(obj)
        list.add(obj)
    }

    override fun getCount(): Int {
        return list.size
    }

    override fun getItem(position: Int): Any? {
        return list[position]
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var row = convertView
        val contactHolder: ContactHolder
        if (row == null) {
            val layoutInflater = this.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            row = layoutInflater.inflate(R.layout.row_layout, parent, false)
            contactHolder = ContactHolder()
            contactHolder.Item = row.findViewById(R.id.item)
            row.tag = contactHolder
        } else {
            contactHolder = row.tag as ContactHolder
        }
        val contacts = this.getItem(position) as Contacts
        contactHolder.Item!!.text = contacts.getItem()
        return row!!
    }

    internal class ContactHolder {
        var Item: TextView? = null
    }
}
